source/compatibility.tex

```diff
@@ -229,7 +229,7 @@
 The \tcode{shrink_to_fit} member function can be used instead.
 \effect
 A valid \CppXXIII{} program that calls \tcode{reserve()}
-on a \tcode{basic_string} object may fail to compile.
+on a \tcode{basic_string} object may become ill-formed.
```

Comment:
Is it suitable to use "may" here (and possibly elsewhere)? Doesn't the program definitely become ill-formed?

Reply:
Yes, it might fail to compile, but it might still be accepted by a given implementation as an extension. However, it's definitely ill-formed ... but might still compile with a given implementation. 

I don't think this change is as simple as replacing "fail to compile" with "become ill-formed". The existing wording is a bit loose but that reflects the reality of what happens in practice.

Reply:
I think the cases that "might fail to compile or might change behaviour" are fine to change this way. The ones that can't change behaviour just become ill-formed, no there's no "may". (Implementations are still permitted to accept ill-formed programs with a diagnostic if they want to.)